Documentation/Getting-Started-with-Modules-Part-1.markdown

```diff
@@ -0,0 +1,427 @@
+
+  1. Type ` : ContentPart` after your `FeaturedProductPart` class definition to inherit from the `ContentPart` class.
+  
+  1. Wait a second and the red squiggles will appear underneath the class. Add the namespace by pressing `Ctrl-.` on your keyboard to bring up the Quick Actions menu.
+  
+  1. Select the `using Orchard.ContentManagement;` option and press `enter`.
+    
+ That's all you need to do for your first ContentPart class.
+
+## Data migrations
+When your module is enabled in the admin dashboard the the data migration class runs. The purpose of the data migration class is to register the module with Orchard and list the features and data it uses.
+
+We aren't going to use this yet, but the migration is also used for upgrades. As you work on your modules you will want to add and remove bits. The data migration class can make changes and you can transform your existing data to meet your new requirements.
+
+The data migration class can be created by hand, following a similar process as the last section or we can scaffold it with the `orchard.exe` command line. Let's dive back in to the command line and add a data migration class to the module.
+
+  1. Press the `Save All` button (or press `Ctrl-Shift-S`). Its a good practice to always save before using the command line utility. Many of its commands will make changes to your solution and if you have unsaved changes you will get merge conflicts.
+  
+  1. In the Solution Explorer, scroll down to the `Orchard.Web` project.
+  
+  1. Right click on `Orchard.Web` project and choose `Open Folder in File Explorer`:
+     
+     ![](../Attachments/getting-started-with-modules-part-1/open-orchard-web.png)
+  
+  1. Open the `bin` folder.
+  
+  1. Locate `orchard.exe` in the list and double click it to open.
+  
+     ![](../Attachments/getting-started-with-modules-part-1/open-orchard-exe.png)
+  
+  1. After a short pause while it loads you will then be presented with the Orchard command line:
+  
+     ![](../Attachments/getting-started-with-modules-part-1/orchard-command-line.png)
+      
+     There is a separate article where you can [learn more about orchard.exe and its features](Using-the-command-line-interface).
+  
+  1. We enabled the code generation feature when scaffolding the module but if you have been playing with Orchard or are just using this guide as a reference it can't hurt to run the command a second time to make sure. 
+  
+     Type the following command: `feature enable Orchard.CodeGeneration` and press `enter`.
+  
+     ![](../Attachments/getting-started-with-modules-part-1/enable-codegeneration.png)
+     
+     This will activate the code generation features of `orchard.exe`. The command that we will be using is `codegen datamigrations`. 
+     
+  1. Type `help codegen datamigrations` and press enter to see the syntax for this command. To see details about all of the commands available type `help commands`. 
```

Comment:
The 'help' command didn't work, but codeen worked perfectly, it might have something to do with the Orchard version?? What Orchard version did you use to write the tutorial?
![learningorchard1](https://cloud.githubusercontent.com/assets/4906291/10416034/d6c1d252-7004-11e5-840c-d3f2d1a5f08d.PNG)